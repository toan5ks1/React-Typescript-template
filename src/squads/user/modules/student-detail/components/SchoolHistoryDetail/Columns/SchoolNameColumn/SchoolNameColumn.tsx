import { memo } from "react";

import Box from "@mui/material/Box";
import TypographyBase from "src/components/Typographys/TypographyBase";

export interface SchoolNameColumnProps {
    content: string;
    isCurrentSchool: boolean;
}

function SchoolNameColumn({ content, isCurrentSchool }: SchoolNameColumnProps) {
    return (
        <Box display="flex" alignItems="center">
            {isCurrentSchool && (
                <TypographyBase
                    sx={(theme) => ({
                        color: theme.palette.success.main,
                        pr: theme.spacing(0.5),
                    })}
                    variant="body2"
                >
                    [Current]
                </TypographyBase>
            )}

            <TypographyBase variant="body2">{content}</TypographyBase>
        </Box>
    );
}

export default memo(SchoolNameColumn);
