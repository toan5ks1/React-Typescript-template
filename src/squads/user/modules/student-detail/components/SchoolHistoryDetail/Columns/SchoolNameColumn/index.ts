export { default } from "./SchoolNameColumn";
export * from "./SchoolNameColumn";
