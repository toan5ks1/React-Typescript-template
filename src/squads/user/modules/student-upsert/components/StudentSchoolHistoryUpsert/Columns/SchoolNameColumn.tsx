import { memo } from "react";

import AutocompleteTooltipHF from "src/squads/user/components/Autocomplete/AutocompleteTooltipHF";

import { SchoolHistoryColumnsProps } from "./types";

interface SchoolNameColumnProps extends SchoolHistoryColumnsProps {
    name: string;
}

function SchoolNameColumn(props: SchoolNameColumnProps) {
    return (
        <AutocompleteTooltipHF
            options={[]}
            name={props.name}
            optionLabelKey={""}
            getOptionSelectedField={""}
        />
    );
}

export default memo(SchoolNameColumn);
