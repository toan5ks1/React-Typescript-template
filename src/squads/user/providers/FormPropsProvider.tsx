import { createContext, PropsWithChildren, useContext } from "react";

export interface FormPropsContextValue {
    readOnly: boolean;
}

const FormPropsContext = createContext<FormPropsContextValue>({ readOnly: false });

const { Provider } = FormPropsContext;

export const FormPropsProvider = ({
    children,
    readOnly,
}: PropsWithChildren<{ readOnly: boolean }>) => {
    return <Provider value={{ readOnly }}>{children}</Provider>;
};

export default FormPropsContext;
export const useFormProps = () => useContext(FormPropsContext);
