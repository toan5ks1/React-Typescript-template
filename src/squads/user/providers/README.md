### We abstract the provider into an isolated component, can be hard to test but parent component don't need to care about what the provider does
