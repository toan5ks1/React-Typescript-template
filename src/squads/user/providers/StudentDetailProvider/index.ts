export * from "./StudentDetailProvider";
export { default } from "./StudentDetailProvider";
