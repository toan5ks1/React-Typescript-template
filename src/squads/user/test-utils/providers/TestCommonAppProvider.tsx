import { PropsWithChildren } from "react";

import { TestContext } from "ra-test";

import TranslationProvider from "src/squads/user/providers/TranslationProvider";

import TestThemeProvider from "./TestThemeProvider";

const TestCommonAppProvider = ({ children }: PropsWithChildren<{}>) => {
    return (
        <TestContext>
            <TranslationProvider>
                <TestThemeProvider>{children}</TestThemeProvider>
            </TranslationProvider>
        </TestContext>
    );
};

export default TestCommonAppProvider;
