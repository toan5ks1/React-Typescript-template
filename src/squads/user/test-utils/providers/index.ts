export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as TestCommonAppProvider } from "./TestCommonAppProvider";
export * from "./TestQueryWrapper";
export { default as TestHookForm } from "./TestHookForm";
export { default as TestHookFormProvider } from "./TestHookFormProvider";
export { default as TestingRouter } from "./TestingRouter";
export { default as AppProvider } from "./AppProvider";
export { default as TestStudentDetailProvider } from "./TestStudentDetailProvider";
