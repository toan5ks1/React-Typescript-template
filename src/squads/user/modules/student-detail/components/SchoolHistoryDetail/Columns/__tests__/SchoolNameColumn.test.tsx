import { TestCommonAppProvider } from "src/squads/user/test-utils/providers";

import SchoolNameColumn, { SchoolNameColumnProps } from "../SchoolNameColumn";

import { render, screen } from "@testing-library/react";

describe("<CommonColumn/>", () => {
    const renderComponent = (props?: Partial<SchoolNameColumnProps>) => {
        return render(
            <TestCommonAppProvider>
                <SchoolNameColumn content="Test" isCurrentSchool {...props} />
            </TestCommonAppProvider>
        );
    };

    it("should match snapshot", () => {
        const { container } = renderComponent();
        expect(container).toMatchSnapshot();
    });

    it("should render content and current school props", () => {
        renderComponent();
        expect(screen.getByText("Test")).toBeInTheDocument();
        expect(screen.getByText("Test")).toHaveStyle("font-size:0.875rem"); //  14px ===  0.875rem
        expect(screen.getByText("[Current]")).toBeInTheDocument();
        expect(screen.getByText("[Current]")).toHaveStyle("color:rgb(76, 175, 80)"); // rgb(76, 175, 80) === "#4CAF50"
    });

    it("should not render current school", () => {
        renderComponent({ isCurrentSchool: false });

        expect(screen.queryByText("[Current]")).not.toBeInTheDocument();
    });
});
