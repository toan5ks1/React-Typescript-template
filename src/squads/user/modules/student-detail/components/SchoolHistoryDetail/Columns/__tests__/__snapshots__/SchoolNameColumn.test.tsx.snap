// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommonColumn/> should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
  color: #4CAF50;
  padding-right: 4px;
}

.emotion-2 {
  margin: 0;
  font-size: 0.875rem;
  font-weight: 400;
  letter-spacing: 0.15px;
  text-transform: none;
  font-family: Roboto,sans-serif;
  line-height: 1.43;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-1"
    >
      [Current]
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-2"
    >
      Test
    </p>
  </div>
</div>
`;
